lasy/profiles/transverse/gaussian_profile.py

```diff
@@ -19,11 +19,28 @@ class GaussianTransverseProfile(TransverseProfile):
     ----------
     w0 : float (in meter)
         The waist of the laser pulse, i.e. :math:`w_0` in the above formula.
+
+    wavelength : float (in meter)
+        The main laser wavelength :math:`\\lambda_0` of the laser.
+
+    z_foc : float (in meter), optional
+        Position of the focal plane. (The laser pulse is initialized at `z=0`.)
+
+    .. warning::
+
+        In order to initialize the pulse out of focus, you can either:
+            - Use a non-zero `z_foc`
+            - Use `z_foc=0` (i.e. initialize the pulse at focuse) and then
+              call `laser.propagate(-z_foc)`
+        Both methods are in principle equivalent, but note that the first
+        method uses the paraxial approximation, while the second method does
+        not make this approximation.
     """
 
-    def __init__(self, w0):
+    def __init__(self, w0, wavelength, z_foc=0):
```

Comment:
Should we make `wavelength` optional and check that it is mentioned only if `z_foc` is specified (or non-zero)? I believe this could help consistency with other profiles. No strong opinion.

Reply:
As discussed offline, let's merge as is and fix this later.